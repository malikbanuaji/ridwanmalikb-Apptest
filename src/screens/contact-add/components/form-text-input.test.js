import * as React from 'react';

import {render} from '@testing-library/react-native';
import FormTextInput from './form-text-input';

test('should render properly', () => {
  const props = {
    onBlur: jest.fn(),
    onChangeText: jest.fn(),
    value: '',
    label: 'Label',
    placeholder: 'Placeholder',
    isError: false,
    keyboardType: undefined,
  };
  const component = render(<FormTextInput {...props} />);
  expect(component.toJSON()).toMatchSnapshot();
});

test('should render properly with error', () => {
  const props = {
    onBlur: jest.fn(),
    onChangeText: jest.fn(),
    value: '',
    label: 'Label',
    placeholder: 'Placeholder',
    isError: true,
    keyboardType: undefined,
  };
  const component = render(<FormTextInput {...props} />);
  expect(component.toJSON()).toMatchSnapshot();
});
