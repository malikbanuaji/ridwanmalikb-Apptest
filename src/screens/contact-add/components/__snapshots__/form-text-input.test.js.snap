// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly 1`] = `
<View>
  <Text
    style={
      Object {
        "color": "#99aab5",
        "fontSize": 14,
      }
    }
  >
    Label
  </Text>
  <TextInput
    onBlur={[MockFunction]}
    onChangeText={[MockFunction]}
    placeholder="Placeholder"
    placeholderTextColor="#6c757a"
    style={
      Object {
        "borderBottomColor": "#99aab5",
        "borderBottomWidth": 1,
        "color": "#ffffff",
        "fontSize": 16,
        "paddingHorizontal": 0,
        "paddingVertical": 5,
      }
    }
    value=""
  />
</View>
`;

exports[`should render properly with error 1`] = `
<View>
  <Text
    style={
      Object {
        "color": "#99aab5",
        "fontSize": 14,
      }
    }
  >
    Label
  </Text>
  <TextInput
    onBlur={[MockFunction]}
    onChangeText={[MockFunction]}
    placeholder="Placeholder"
    placeholderTextColor="#6c757a"
    style={
      Object {
        "borderBottomColor": "#f57731",
        "borderBottomWidth": 1,
        "color": "#ffffff",
        "fontSize": 16,
        "paddingHorizontal": 0,
        "paddingVertical": 5,
      }
    }
    value=""
  />
</View>
`;
